cmd/frontend/graphqlbackend/textsearch.go

```diff
@@ -44,7 +44,7 @@ type FileMatchResolver struct {
 	JLimitHit    bool         `json:"LimitHit"`
 	MatchCount   int          // Number of matches. Different from len(JLineMatches), as multiple lines may correspond to one logical match.
 	symbols      []*searchSymbolResult
-	uri          string
+	URI          string `json:"-"`
```

Comment:
For JSON, I ignored the field so that JSON encoding behavior won't change.

Reply:
After looking at [schema.graphql](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/schema.graphql) I don't thinkthe `json:"-"` is needed. Did you notice anything break if this is omitted?

Reply:
I hadn't tested it without -- I was doing it preventatively. I don't have a great understanding of how the schema works. Is there generated code somewhere that's not using that field?

Reply:
>  how the schema works

My understanding is that the resolver types will export to the schema fields if the struct field names match (either plainly, like `MatchCount`, or with the JSON export tag like `json:"LimitHit"`. I _think_ everything else is ignored, and it looks like `uri` isn't part of the FileMatch resolver (I thought before maybe it was). In any case, if `uri` _were_ exported before, and now we ignore it with `"-"`, it would break the interface. This can only reasonably be 

`uri string` (as-is)

```
uri string `json:"uri"` // if uri is exported for this resolver, which I believe it is not.
``` 

So let's go with:

```suggestion
	URI
```

Reply:
Sounds good, but I don't think this could break any reflection-based unmarshalling just because it's impossible for an external package to read an unexported field (same reason that the JSON package can't unmarshal to unexported fields).

Reply:
Oh, good point, I see what you mean! Hmm, so maybe we do need to `json:"-"`. I'll let @keegancsmith chip in from here, looks good to me!

Reply:
If the tests pass it should be good to go :) Our graphql resolver framework only reflects on the methods of a struct, not the fields.